lib/foreman/dynflow/configuration.rb

```diff
@@ -0,0 +1,152 @@
+module Foreman
+  class Dynflow::Configuration
+    # the number of threads in the pool handling the execution
+    attr_accessor :pool_size
+
+    # the size of db connection pool
+    attr_accessor :db_pool_size
+
+    # set true if the executor runs externally (by default true in procution, othewise false)
+    attr_accessor :remote
+    alias remote? remote
+
+    # what transaction adapater should be used, by default, it uses the ActiveRecord
+    # based adapter, expecting ActiveRecord is used as ORM in the application
+    attr_accessor :transaction_adapter
+
+    attr_accessor :eager_load_paths
+
+    attr_accessor :lazy_initialization
+
+    # what rake tasks should run their own executor, not depending on the external one
+    attr_accessor :rake_tasks_with_executor
+
+    # if true, the ForemanTasks::Concerns::ActionTriggering will make
+    # no effect. Useful for testing, where we mignt not want to execute
+    # the orchestration tied to the models.
+    attr_accessor :disable_active_record_actions
+
+    def initialize
+      self.pool_size                = 5
```

Comment:
Just took the default from foreman_tasks to be honest, I have not experimented with different sizes (lame excuse, I know)